src/codewhisperer/util/closingBracketUtil.ts

```diff
@@ -110,20 +190,87 @@ export const removeBracketsFromRightContext = async (
     }
 }
 
+/**
+ * @param endPosition: end position of the recommendation
+ * @param startPosition: start position of the recommendation
+ */
 export async function handleExtraBrackets(
     editor: vscode.TextEditor,
     recommendation: string,
-    endPosition: vscode.Position
+    endPosition: vscode.Position,
+    startPosition: vscode.Position
 ) {
     const end = editor.document.offsetAt(endPosition)
+    const leftContext = editor.document.getText(
+        new vscode.Range(
+            startPosition,
+            editor.document.positionAt(
+                Math.max(editor.document.offsetAt(startPosition) - CodeWhispererConstants.charactersLimit, 0)
+            )
+        )
+    )
+
     const rightContext = editor.document.getText(
         new vscode.Range(
             editor.document.positionAt(end),
             editor.document.positionAt(end + CodeWhispererConstants.charactersLimit)
         )
     )
-    const bracketsToRemove = getBracketsToRemove(recommendation, rightContext)
+    const bracketsToRemove = getBracketsToRemove(
+        editor,
+        recommendation,
+        leftContext,
+        rightContext,
+        endPosition,
+        startPosition
+    )
     if (bracketsToRemove.length) {
         await removeBracketsFromRightContext(editor, bracketsToRemove, endPosition)
     }
 }
+
+function isPairedSymbol(char1: string, char2: string) {
+    return isPairedParenthesis(char1, char2) || isPairedQuotes(char1, char2)
+}
+
+function isPairedParenthesis(char1: string, char2: string) {
```

Comment:
This is getting to be an extreme amount of code for non-trivial but very common text-editing operations. I.e. it's time to consider looking at a NPM library that can do these calculations instead of re-inventing them.

Reply:
@justinmk3 I will take this as a follow up, let me create a tt as CW tech debt and we will be tracking there

Reply:
since this issue is broadly raised by customers, we want to roll out this ASAP lol